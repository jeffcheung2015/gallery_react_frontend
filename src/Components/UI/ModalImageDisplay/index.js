import React, { Component } from 'react';
import "./index.scss"
import { pure } from 'recompose';
import Loader from 'react-loaders'
import { Modal, Table, Image, Icon, Label } from 'semantic-ui-react'
import PropTypes from 'prop-types'
import _map from 'lodash/map'

const ModalImageDisplay = (props) => {
  const{
    handleModalClose,
    imgPreviewOpen, currImgName, currImgDesc, currImgCreatedAt, currTags, tags, currImgSrc
  } = props
  return (
    <Modal
      className="Modal-imageViewer"
      open={imgPreviewOpen}
      onClose={handleModalClose}
      >
      <Table basic='very' celled collapsing className="Table-imgDescContent">
      <Table.Body>
        <Table.Row>
          <Table.Cell className="TableCell-imgDescContent-key"><Icon name="list"/>Name:</Table.Cell>
          <Table.Cell className="TableCell-imgDescContent-val">{currImgName}</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell className="TableCell-imgDescContent-key"><Icon name="newspaper outline"/>Description:</Table.Cell>
          <Table.Cell className="TableCell-imgDescContent-val">{currImgDesc}</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell className="TableCell-imgDescContent-key"><Icon name="calendar alternate outline"/>Uploaded Date:</Table.Cell>
          <Table.Cell className="TableCell-imgDescContent-val">{currImgCreatedAt}</Table.Cell>
        </Table.Row>
        <Table.Row>
          <Table.Cell className="TableCell-imgDescContent-key"><Icon name="tags"/>Tags:</Table.Cell>
          <Table.Cell className="TableCell-imgDescContent-val">
          {
            _map(currTags, (elem, idx) => {
              return (
                <Label className="Label-tags" key={"currLabel-" + idx}>
                  {tags[currTags[idx]-1]}
                </Label>
              )
            })
          }
          </Table.Cell>
          </Table.Row>
        </Table.Body>
      </Table>
      <Image alt='imageViewer-img' src={process.env.REACT_APP_SERVER_HOST_URL + currImgSrc} wrapped />
    </Modal>
  );
}
export default pure(ModalImageDisplay);

ModalImageDisplay.propTypes = {
  currImgName:PropTypes.string.isRequired,
  currImgSrc:PropTypes.string.isRequired,
  currImgDesc:PropTypes.string.isRequired,
  currImgCreatedAt:PropTypes.string.isRequired,
  currTags:PropTypes.arrayOf(PropTypes.number).isRequired,
  handleModalClose:PropTypes.func.isRequired,
  imgPreviewOpen:PropTypes.bool.isRequired,
}
