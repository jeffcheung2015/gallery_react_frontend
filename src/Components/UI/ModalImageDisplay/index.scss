.Modal-imageViewer{
  width: auto !important;
  display: inline-flex;
  max-width: 40%;
  .Table-imgDescContent{
    position: relative;
    display: grid;
    border-radius: 5px;
    padding: 20px;
    position: absolute;
    right: 100%;
    min-width: 350px;
    .TableCell-imgDescContent-key, .TableCell-imgDescContent-val{
      color: #FFFFFF;
      border-color: #FFFFFF;
      border-left-color: #FFFFFF !important;
      .Label-tags{
        margin: 5px;
      }
    }
    .TableCell-imgDescContent-key{
      display: flex;
    }
  }
}
