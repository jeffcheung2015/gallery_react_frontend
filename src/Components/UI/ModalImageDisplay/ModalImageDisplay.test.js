import React, { Component } from 'react'
import _times from 'lodash/times'
import renderer from 'react-test-renderer';
import ModalImageDisplay from 'Components/UI/ModalImageDisplay';
import toJson from 'enzyme-to-json'
let [handleModalClose] = _times(1,() => jest.fn())

let wrapper

let fixtureTags = ["Cat", "Dog", "Animals", "Flowers", "Scenery", "Monkey", "Mammals", "Lion", "Human", "Rabbit", "Gorilla", "Bird", "Owl", "Pig", "Fish", "Bear", "Reptile", "Insect", "Sponge", "Elephant", "Whale", "Shark", "Penguin", "Wolf", "Turtle", "Mouse", "Goose", "Kangaroo"]

let dataProps = {imgPreviewOpen: false, currImgName: "zebra", currImgDesc: 'dummy desc',
 currImgCreatedAt: '2018-06-12 09:55:22', currTags : [2,3,4], tags: fixtureTags, currImgSrc: "/api/zebra.jpg"}
let funcProps = {handleModalClose}

describe("<ModalImageDisplay />", () => {
  it("input -> correct output test", () => {
    const tree = shallow(
      <ModalImageDisplay
        {...dataProps}
        {...funcProps}
      />
    ).dive()
    expect(toJson(tree)).toMatchSnapshot();
  })
})
