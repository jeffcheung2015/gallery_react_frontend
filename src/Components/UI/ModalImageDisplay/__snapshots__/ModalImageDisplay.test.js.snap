// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalImageDisplay /> input -> correct output test 1`] = `
<Modal
  centered={true}
  className="Modal-imageViewer"
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[MockFunction]}
  open={false}
>
  <Table
    as="table"
    basic="very"
    celled={true}
    className="Table-imgDescContent"
    collapsing={true}
  >
    <TableBody
      as="tbody"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableCell
          as="td"
          className="TableCell-imgDescContent-key"
        >
          <Icon
            as="i"
            name="list"
          />
          Name:
        </TableCell>
        <TableCell
          as="td"
          className="TableCell-imgDescContent-val"
        >
          zebra
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableCell
          as="td"
          className="TableCell-imgDescContent-key"
        >
          <Icon
            as="i"
            name="newspaper outline"
          />
          Description:
        </TableCell>
        <TableCell
          as="td"
          className="TableCell-imgDescContent-val"
        >
          dummy desc
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableCell
          as="td"
          className="TableCell-imgDescContent-key"
        >
          <Icon
            as="i"
            name="calendar alternate outline"
          />
          Uploaded Date:
        </TableCell>
        <TableCell
          as="td"
          className="TableCell-imgDescContent-val"
        >
          2018-06-12 09:55:22
        </TableCell>
      </TableRow>
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableCell
          as="td"
          className="TableCell-imgDescContent-key"
        >
          <Icon
            as="i"
            name="tags"
          />
          Tags:
        </TableCell>
        <TableCell
          as="td"
          className="TableCell-imgDescContent-val"
        >
          <Label
            className="Label-tags"
            key="currLabel-0"
          >
            Dog
          </Label>
          <Label
            className="Label-tags"
            key="currLabel-1"
          >
            Animals
          </Label>
          <Label
            className="Label-tags"
            key="currLabel-2"
          >
            Flowers
          </Label>
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
  <Image
    alt="imageViewer-img"
    as="img"
    src="undefined/api/zebra.jpg"
    ui={true}
    wrapped={true}
  />
</Modal>
`;
